Commits of FT2025-2.

blchoy opened the pull request:
Commits of FT2025-2.

blchoy committed:
First commit of FT2025-2.

amilan17 commented on CSV/306/306_container.csv:
```diff
@@ -1,2 +1,2 @@
-id,notation,status,description,label,altLabel,modified,versionInfo,seeAlso,manager,owner
-http://codes.wmo.int/306/4678,4678,stable,"This register provides terms that may be used to describe significant [recent], present and forecast weather - enumerating the set of meteorologically valid options defined in WMO No. 306 Vol 1 code-table 4678. Each term comprises one or more of the elements from code-table 4678: 'intensity or proximity', 'descriptor', 'precipitation', 'obscuration' and  ...
+id,notation,status,description,label,altLabel,modified,source,seeAlso,publisher,manager,owner,note,related,iwxxmVersionInfo
```
I don't think we need columns for modified, publisher, manager or owner

blchoy replied:
Agree to remove "publisher", "manager" and "owner" from the CSV files.  I am not too sure for "modified".  I understand it is auto-generated by the server during inserting/updating and is more meaningful for server operation then other purposes.  What I would like to do is to record when a registry is published, as this is a piece of important information which is hard to reconstruct from change notes.  May be we also remove "modified" from the CSV files and introduce a new column "Publish date" to indicate when a registry is published?  Entries in this column will only be changed when there is a new registry or updates to an existing registry published, which is (taking 2025-2 as an example) the publishing date of 2025-2RC1, 2025-2RC2 and the approved version 2025-2.

We can leave "Publish date" in the CSV files only as there may not be value to have in either the WMO Codes Register or the RDF files for validation.

amilan17 replied:
Modified is added/updated by the application. I'm ok with using a publish date in the CSV, but I feel that the version is sufficient enough in capturing that information...

blchoy replied:
> but I feel that the version is sufficient enough in capturing that information...

May be true for the second FY of the year (round about Nov) but there is no guarantee.  For Release Candidates there is definitely no way to guess.

blchoy replied:
Changes (delete columns "publisher", "manager", "owner" and rename "modified" to "publishDate") have been applied to all _container.csv files.

amilan17 commented on CSV/306/306_container.csv:
```diff
@@ -1,2 +1,2 @@
-id,notation,status,description,label,altLabel,modified,versionInfo,seeAlso,manager,owner
-http://codes.wmo.int/306/4678,4678,stable,"This register provides terms that may be used to describe significant [recent], present and forecast weather - enumerating the set of meteorologically valid options defined in WMO No. 306 Vol 1 code-table 4678. Each term comprises one or more of the elements from code-table 4678: 'intensity or proximity', 'descriptor', 'precipitation', 'obscuration' and  ...
+id,notation,status,description,label,altLabel,modified,source,seeAlso,publisher,manager,owner,note,related,iwxxmVersionInfo
+http://codes.wmo.int/306/4678,4678,stable,"This register provides terms that may be used to describe significant [recent], present and forecast weather - enumerating the set of meteorologically valid options defined in WMO No. 306 Vol 1 code-table 4678. Each term comprises one or more of the elements from code-table 4678: 'intensity or proximity', 'descriptor', 'precipitation', 'obscuration' and  ...
```
```suggestion
http://codes.wmo.int/306/4678,4678,stable,"This register provides terms that may be used to describe significant [recent], present and forecast weather - enumerating the set of meteorologically valid options defined in WMO No. 306 Vol 1 code-table 4678. Each term comprises one or more of the elements from code-table 4678: 'intensity or proximity', 'descriptor', 'precipitation', 'obscuration' and 'other'.",Code Table 4678: Significant weather phenomena,w'w',2017-09-15T15:45:38.696Z,,https://library.wmo.int/idurl/4/35713,http://codes.wmo.int/system/organization/wmo,http://codes.wmo.int/system/organization/www-dm,http://codes.wmo.int/system/organization/wmo,,,1.1;2.1;3.0;2021-2;2023-1;2025-2
```
```suggestion
http://codes.wmo.int/306/4678,4678,stable,"This register provides terms that may be used to describe significant [recent], present and forecast weather - enumerating the set of meteorologically valid options defined in WMO No. 306 Vol 1 code-table 4678. Each term comprises one or more of the elements from code-table 4678: 'intensity or proximity', 'descriptor', 'precipitation', 'obscuration' and 'other'.",Code Table 4678: Significant weather phenomena,w'w',2017-09-15T15:45:38.696Z,,http://www.wmo.int/pages/prog/www/WMOCodes/WMO306_vI1/VolumeI.1.html,http://codes.wmo.int/system/organization/wmo,http://codes.wmo.int/system/organization/www-dm,http://codes.wmo.int/system/organization/wmo,,,1.1;2.1;3.0;2021-2;2023-1;2025-2
```

blchoy replied:
Is the link https://library.wmo.int/idurl/4/35713 stable too?  My gut feeling is that historical documents will eventually disappear from online sources and we will need to update the record at that time.

amilan17 replied:
The WMO library recently added these URIs for all the publications and they will not change.

blchoy replied:
Changes made to 306_container.csv.

blchoy replied:
I have made the changes as indicated.  I have also located those WMO No.49-2 and changed them to Annex 3 + PANS-MET too.

amilan17 commented on CSV/49-2/49-2_container.csv:
```diff
@@ -1,14 +1,15 @@
+http://codes.wmo.int/49-2/SpaceWxLocation,SpaceWxLocation,stable,The items within this code table are the locations of space weather phenomena of significance to aeronautical operations,Space Wx Location,,2019-03-20T16:57:02.324Z,http://codes.wmo.int/system/organization/wmo,http://codes.wmo.int/system/organization/www-dm,http://codes.wmo.int/system/organization/wmo,,,3.0;2021-2;2023-1;2025-2
+http://codes.wmo.int/49-2/SpaceWxPhenomena,SpaceWxPhenomena,stable,The items within this code table are the types of space weather phenomena of significance to aeronautical operations,Space Wx Phenomena,,2019-03-20T16:54:58.099Z,http://codes.wmo.int/system/organization/wmo,http://codes.wmo.int/system/organization/www-dm,http://codes.wmo.int/system/organization/wmo,,,3.0;2021-2;2023-1
+http://codes.wmo.int/49-2/WeatherCausingVisibilityReduction,WeatherCausingVisibilityReduction,stable,ICAO Annex 3/ WMO No. 49-2 Appendix 6 Section 2.1 under SFC VIS,Weather Causing Visibility Reduction,,2016-12-30T13:26:25.95Z,http://codes.wmo.int/system/organization/wmo,http://codes.wmo.int/system/organization/www-dm,http://codes.wmo.int/system/organization/wmo,,,2.1;3.0;2021-2;2023-1;2025-2
+http://codes.wmo.int/49-2/observable-property,observable-property,stable,"The items within this code table are composite observable properties that define the set of physical properties evaluated as a result of regulated procedures specified in Technical Regulations, Volume II (WMO-No. 49): Meteorological Services for International Air Navigation. These include aerodrome observation and forecast  ...
```
```suggestion
http://codes.wmo.int/49-2/observable-property,observable-property,stable,"The items within this code table are composite observable properties that define the set of physical properties evaluated as a result of regulated procedures specified in Technical Regulations, Volume II (WMO-No. 49): Meteorological Services for International Air Navigation. These include aerodrome observation and forecast reports (e.g. METAR, SPECI and TAF).",Code Table D-5: IWXXM observable properties,,2017-11-23T11:08:18.808Z,http://codes.wmo.int/system/organization/wmo,http://codes.wmo.int/system/organization/www-dm,http://codes.wmo.int/system/organization/wmo,,D-5,1.1;2.1
```
```suggestion
http://codes.wmo.int/49-2/observable-property,observable-property,stable,"The items within this code table are composite observable properties that define the set of physical properties evaluated as a result of regulated procedures specified in Technical Regulations, Volume II (WMO-No. 49): Meteorological Services for International Air Navigation. These include aerodrome observation and forecast reports (e.g. METAR, SPECI and TAF). http://codes.wmo.int/system/organization/wmo",Code Table D-5: IWXXM observable properties,,2017-11-23T11:08:18.808Z,http://codes.wmo.int/system/organization/wmo,http://codes.wmo.int/system/organization/www-dm,http://codes.wmo.int/system/organization/wmo,,D-5,1.1;2.1
```

blchoy replied:
This involves removal of "owner" which has been made in previous suggestion.

amilan17 commented on CSV/49-2/AerodromePresentOrForecastWeather/AerodromePresentOrForecastWeather_entity.csv:
```diff
@@ -1,403 +1,403 @@
-http://codes.wmo.int/306/4678/VCSS,VCSS,stable,Sandstorm in the vicinity,Sandstorm in the vicinity,http://codes.wmo.int/306/4678/VCSS
-http://codes.wmo.int/306/4678/VCTS,VCTS,stable,Thunderstorm in the vicinity,Thunderstorm in the vicinity,http://codes.wmo.int/306/4678/VCTS
-http://codes.wmo.int/306/4678/VCVA,VCVA,stable,Volcanic ash in the vicinity,Volcanic ash in the vicinity,http://codes.wmo.int/306/4678/VCVA
+id,notation,status,description,label,altLabel,source,seeAlso,related,note,iwxxmVersionInfo
```
the related URLs are identical to the id URLs

blchoy replied:
Ah... both 49-2/AerodromePresentOrForecastWeather and 49-2/AerodromeRecentWeather are "virtual" registers and their are essentially restricted sets of entries in 306/4678:

```
  <reg:Register rdf:about="http://codes.wmo.int/49-2/AerodromePresentOrForecastWeather">
    <skos:member>
      <skos:Concept rdf:about="http://codes.wmo.int/306/4678/PLDZRA">
        <rdfs:label xml:lang="en">Precipitation of ice pellets, drizzle and rain</rdfs:label>
      </skos:Concept>
    </skos:member>
    <skos:member>
      <skos:Concept rdf:about="http://codes.wmo.int/306/4678/FZDZ">
        <rdfs:label xml:lang="en">Precipitation of freezing drizzle</rdfs:label>
      </skos:Concept>
    </skos:member>
```

I agree that since the ids are already in 306/4678 there is no need to specify it again in related URL.

blchoy replied:
The "related" column has been emptied.

amilan17 commented on CSV/49-2/SpaceWxLocation/SpaceWxLocation_entity.csv:
```diff
@@ -1,8 +1,10 @@
+http://codes.wmo.int/49-2/SpaceWxLocation/DAYLIGHT_SIDE,DAYLIGHT_SIDE,stable,Daylight Side,Daylight Side,,,,,,3.0;2021-2;2023-1
+http://codes.wmo.int/49-2/SpaceWxLocation/DAYSIDE,DAYSIDE,experimental,Day Side,Day Side,,,,,,2025-2
+http://codes.wmo.int/49-2/SpaceWxLocation/NIGHTSIDE,NIGHTSIDE,experimental,Night Side,Night Side,,,,,,2025-2
+http://codes.wmo.int/49-2/SpaceWxLocation/EQN,EQN,stable,Equatorial latitudes northern hemisphere (30\u00B0N - 0\u00B0N),Equatorial latitudes northern hemisphere,,,,,,3.0;2021-2;2023-1;2025-2
```
```suggestion
http://codes.wmo.int/49-2/SpaceWxLocation/EQN,EQN,stable,Equatorial latitudes northern hemisphere (30 - 0),Equatorial latitudes northern hemisphere,,,,,,3.0;2021-2;2023-1;2025-2
```

amilan17 replied:
remove non-ascii chars in this line and below...

blchoy replied:
OK.  Shall we say 30N or 30 degrees N or otherwise?  We may also have that kind of issue in the schema notes.

amilan17 replied:
I think "30 degrees N" is the best approach

blchoy replied:
Changes made.

amilan17 commented on CSV/49-2/VolcanicAshConcentrations/VolcanicAshConcentrations_entity.csv:
```diff
@@ -0,0 +1,5 @@
+id,notation,status,description,label,altLabel,source,seeAlso,related,note,iwxxmVersionInfo
+http://codes.wmo.int/49-2/VolcanicAshConcentrations/LOW_AND_POSSIBLY_HIGHER,LOW_AND_POSSIBLY_HIGHER,experimental,Greater than or equal to 0.2 mg/m3 - Low and possibly higher,Greater than or equal to 0.2 mg/m3 - Low and possibly higher,,,,,,2025-2
```
```suggestion
http://codes.wmo.int/49-2/VolcanicAshConcentrations/LOW_AND_POSSIBLY_HIGHER,LOW_AND_POSSIBLY_HIGHER,experimental,Greater than or equal to 0.2 mg/m3,Low and possibly higher,,,,,,2025-2
```

amilan17 replied:
divide label and descriptions in this table.

blchoy replied:
Low and possibly higher?  It shouldn't be there anyway as discussed in the meeting.  Let me check the content first.  Agree that descriptions should not be just a copy of label which are not able to give additional information.

amilan17 replied:
> Agree that descriptions should not be just a copy of label which are not able to give additional information.

Yes, where possible for new codes, let's try to NOT repeat the label in the description.

blchoy replied:
Two things:  First of all this code list should not exist under 49-2; it has been moved to iwxxm and it is my mistake for not being removed.  Secondly, it has been decided not to categorize concentration levels at all so this code list should be removed no matter under 49-2 or iwxxm.  See https://github.com/wmo-im/iwxxm/issues/343#issuecomment-2980074169 for details.

amilan17 commented on CSV/49-2/observation-type/observation-type_entity.csv:
Do we need to keep record of subregisters??

blchoy replied:
I don't think we have subregisters so far.  Correct me if I am wrong.

amilan17 replied:
there is a column for this in the observation-type_entity CSV with the content of: "http://codes.wmo.int/49-2/observation-type/IWXXM/2.1>|<http://codes.wmo.int/49-2/observation-type/IWXXM/1.0"

`id,notation,status,description,label,altLabel,source,seeAlso,related,note,iwxxmVersionInfo,modified,publisher,delegationTarget,forwardingCode,manager,owner,subregister
http://codes.wmo.int/49-2/observation-type/IWXXM,,,,ICAO IWXXM observation types,,,,,,1.1;2.1,2017-11-23T11:06:16.632Z,http://codes.wmo.int/system/organization/wmo,,,http://codes.wmo.int/system/organization/www-dm,http://codes.wmo.int/system/organization/wmo,http://codes.wmo.int/49-2/observation-type/IWXXM/2.1>|<http://codes.wmo.int/49-2/observation-type/IWXXM/1.0`

blchoy replied:
> there is a column for this in the observation-type_entity CSV with the content of: "[http://codes.wmo.int/49-2/observation-type/IWXXM/2.1>|](http://codes.wmo.int/49-2/observation-type/IWXXM/2.1%3E%7C)<http://codes.wmo.int/49-2/observation-type/IWXXM/1.0"

That's why I would like to forget "observation-type" (see below).

blchoy replied:
Let me re-iterate: While the CSV files for observable-property and observation-type are there, we will not generate TTL and RDF files and they will not be updated and used by the schematron rules anyway.  Let me know in this circumstances we may also want to not keeping and managing the 2 CSV files under this GitHub repository.

blchoy replied:
I have deleted the two CSV files in the latest update.

amilan17 commented on CSV/49-2/observable-property/observable-property_entity.csv:
```diff
@@ -1,7 +1,7 @@
+http://codes.wmo.int/49-2/observable-property/AIRMETEvolvingConditionCollectionAnalysis,AIRMETEvolvingConditionCollectionAnalysis,stable,"The set of physical properties evaluated as a result of an AIRMET, as specified in Technical Regulations, Volume II (WMO-No. 49): Meteorological Services for International Air Navigation.",AIRMET Evolving Condition Collection Analysis,,,,,,1.1;2.1,,,AIRMET Evol ...
+http://codes.wmo.int/49-2/observable-property/MeteorologicalAerodromeForecast,MeteorologicalAerodromeForecast,stable,"The set of physical properties evaluated as a result of an aerodrome forecast (TAF), as specified in the Technical Regulations (WMO-No. 49), Volume II – Meteorological Service for International Air Navigation.",Meteorological aerodrome forecast,,,,,,1.1;2.1,http://codes.wmo.int/ui ...
+http://codes.wmo.int/49-2/observable-property/MeteorologicalAerodromeObservation,MeteorologicalAerodromeObservation,stable,"The set of physical properties evaluated as a result of the observation procedure of a routine or special aerodrome meteorological report (METAR or SPECI), as specified in the Technical Regulations (WMO-No. 49), Volume II – Meteorological Service for International Air Naviga ...
+http://codes.wmo.int/49-2/observable-property/MeteorologicalAerodromeTrendForecast,MeteorologicalAerodromeTrendForecast,stable,"The set of physical properties evaluated as a result of the trend forecast procedure of a routine or special aerodrome meteorological report (METAR or SPECI), as specified in the Technical Regulations (WMO-No. 49), Volume II – Meteorological Service for International Air ...
```
what is the story behind deletion targets?

blchoy replied:
We discussed at the meeting for not touching 49-2/observable-property, 49-2/observation-type, common/observation-type and common/quantity-kind.  Except for quantity-kind (but it is only being used in annotation of an XML element in the schema, not in an instance) all of them are not being used in IWXXM since version 3.0.  I see no chance for them to be updated and am suggesting to leave them in peace (i.e. not dealing with their associated CSVs, TTL and RDF files).

amilan17 reviewed (changes requested):
see comments, I also recommend changing the order of the columns. 
change name of "id" to "URI". 

notation, label, description are first and URI and status are last... 

Choy: Done

blchoy commented:
> change name of "id" to "URI".

Choy: I have no objection changing from id to URL in CSV files, but when you get an export it shows id and not URL.  Shall we make it more consistent between imports and exports to the WMO Codes Registry?

Anna: I don't think so. As long as the script produces the proper representation in the RDFs and TTLs... 

Choy: Done.

blchoy committed:
Second commit for 2025-2RC2

blchoy committed:
Commit updated build.yml

blchoy committed:
Third commit of 2025-2RC2.  This covers:
1. Further alignments with text changes in the IWXXM schemas
2. Revival of code list AviationColourCode under iwxxm for use with VONA

blchoy committed:
Minor updates to code list AviationColourCode

amilan17 reviewed (changes requested):
Remove URIs from skos:concept in the TTLs  ![image](https://github.com/user-attachments/assets/fe0aef94-f6e0-4bac-a243-499c5635d185)

Choy: Done

amilan17 commented:
> Remove URIs from skos:concept in the TTLs 
The system does not recognize the notation when the concept is URI. 
![image](https://github.com/user-attachments/assets/afbb8a6b-5f3d-4d3c-a207-6c10ca3166c3)

Choy: Done

amilan17 commented on CSV/iwxxm/nil/nil_entity.csv:
do we need to keep all of the nil reason codes in the iwxxm codelist? it looks like only inapplicable and unknown are referenced in the new XSDs...

blchoy replied:
Not right now.  With owl:versionInfo we can make changes later on.

amilan17 replied:
Should we remove the nil reason values that are not in use?

blchoy replied:
Check out https://schemas.wmo.int/iwxxm/2025-2RC1/examples/TAC-to-XML-Guidance.txt and you will find other entities that will be used.  Currently four of the entries are not being used - AboutDectionRange, BelowDetectionRange, template and withheld, but I suspect they will soon be applied in AMOIS, AMFIS and HWIS.  My main concern is on the lack of certain entities discussed previously with Mark, not the list being too long.

So my answer is no.

amilan17 replied:
ok

amilan17 reviewed (commented):
question on nil reasons

amilan17 commented on CSV/49-2/MeteorologicalFeature/MeteorologicalFeature_entity.csv:
Is this referenced in any of the packages?

blchoy replied:
Yes.  All packages which are extensions to Package MeteorologicalFeature use them:

<img src="https://github.com/user-attachments/assets/3ee51f34-ca5e-4a4e-a5ed-50a499b94f07">

amilan17 replied:
@blchoy the screenshot above shows links to the _iwxxm_ codelist, my question is regarding the _49-2_ codelist.

blchoy replied:
49-2/MeteorologicalFeature is being used in current version of IWXXM (i.e. 2023-1), even though no operational data has been published so far.  You reminded me that we should remove 2025-2 in owl:versionInfo of both the registry as well as the entities.  A new version will be committed today.

amilan17 reviewed (commented):
Are we missing http://codes.wmo.int/common/quantity-kind codelist?

Choy: As discussed elsewhere this is not touched now and in future.

blchoy commented:
@amilan17 said:
> Are we missing http://codes.wmo.int/common/quantity-kind codelist?

No.  Those registry entries are only used in the `<annotation>` part of an element in some legacy packages.  This was discussed in the meeting on 13 Jun 2025 and we don't want to touch it:

<img src="https://github.com/user-attachments/assets/b1b44816-6239-4f28-bcd4-968a5d6092a0">

In fact, the export of an entity from the WMO Codes Registry looks different from the others:

```
<http://codes.wmo.int/common/quantity-kind/aerodromeMaximumWindGustSpeed>
        a                qudt:SpaceAndTimeQuantityKind , qudt:QuantityKind , skos:Concept ;
        rdfs:label       "Aerodrome maximum wind gust speed" ;
        <http://codes.wmo.int/def/common/dimensions> "LT-1" ;
        <http://codes.wmo.int/def/common/discipline> "Aeronautical quantity kinds" ;
        dct:description  "Maximum wind speed in the 10 minute period of observation. It is reported only if exceeds the mean speed by 5 m s-1 (10 knots)." ;
        skos:notation    "aerodromeMaximumWindGustSpeed" .
```

I don't think it is worth spending more time on it.  Just let it be.

blchoy committed:
Third commit of 2025-2RC2.

blchoy commented:
> Remove URIs from skos:concept in the TTLs
> The system does not recognize the notation when the concept is URI.

Done.

blchoy committed:
Deprecated the use of 49-2/MeteorologicalFeature commencing 2025-2.

amilan17 reviewed (changes requested):
@blchoy change test register from 
http://testwmocodes.metarelate.net 
to 
https://ci.codes.wmo.int

Choy: Done

amilan17 commented:
@blchoy the upload.py script _cannot_ read the files from the TTL directory. It is expecting the TTLs to be in sister directories of the scripts directory. 

On the other hand, the check_urls.py _can_ read the files from the TTL directory.

Choy: check_urls.py updated to output correctly the location of the TTL files to be uploaded.  A 'upload' subdirectory has also been created to hold outputs from check_urls.py.  I do not have a chance to check upload.py as I cannot go beyond authentication (I have no username and password).

blchoy committed:
Forth commit of 2025-2RC2.

amilan17 commented on upload/output.txt:
@blchoy Thanks for working on this, but I don't think it will work, because the the path to the TTL file has to match the location on the registry. But I can't test it until much later today.

blchoy replied:
What you mean is that the path to TTL files in the output file:

```
{"PUT": [], "POST": ["TTL/iwxxm/AviationColourCode/GREEN.ttl", ... }
```

should be relative paths with host part http://codes.wmo.int, but when used in the upload.py the top level sub-directory TTL/ should be added back when accessing the TTL files on the repository?

Let's work it out during this evening's teams meeting.

blchoy replied:
I have further revised check_urls.py and it should now check the contents too.  I ran it and upload the changes to the test codes registry.  A re-run of check_urls.py indicated that the containers are not properly updated.  The errors of the nil container are as below:

```
======================================================================
FAIL: test_consistent_iwxxm_nil (__main__.TestContentsConsistency.test_consistent_iwxxm_nil)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/mnt/iwxxm-codelists/scripts/check_urls.py", line 163, in entity_consistent
    self.check_result(result_rdfgraph, expected_rdfgraph, uploads, identityURI, resourceURI)
  File "/mnt/iwxxm-codelists/scripts/check_urls.py", line 82, in check_result
    self.assertTrue(rdflib.compare.isomorphic(result, expected),
AssertionError: False is not true : https://ci.codes.wmo.int/iwxxm/nil

####### inBothinputs #######

<http://codes.wmo.int/iwxxm/nil> a <http://purl.org/linked-data/registry#Register>,
        <http://www.w3.org/ns/ldp#Container> .


####### inTestResultOnly #######
@prefix dc: <http://purl.org/dc/elements/1.1/> .
@prefix dcterms: <http://purl.org/dc/terms/> .
@prefix owl: <http://www.w3.org/2002/07/owl#> .
@prefix rdfs: <http://www.w3.org/2000/01/rdf-schema#> .
@prefix skos: <http://www.w3.org/2004/02/skos/core#> .

<http://codes.wmo.int/iwxxm/nil> a skos:Collection ;
    rdfs:label "Nil reasons"@en ;
    dc:source "Code Table D-1: Nil reasons"@en ;
    dcterms:description "Nil-reason terms are used to provide an explanation for recording a missing (or void) value within a data product. Terms are drawn from authorities in addition to WMO including ISO/TC 211 (from ISO 19136:2007 Geography markup language clause 8.2.3.1; published on their behalf by the Open Geospatial Consortium)."@en ;
    owl:versionInfo <http://icao.int/iwxxm/2025-2> ;
    skos:notation "nil" .


####### inExpectedOnly #######
@prefix dc: <http://purl.org/dc/elements/1.1/> .
@prefix dcterms: <http://purl.org/dc/terms/> .
@prefix ns1: <http://purl.org/linked-data/registry#> .
@prefix owl: <http://www.w3.org/2002/07/owl#> .
@prefix rdfs: <http://www.w3.org/2000/01/rdf-schema#> .
@prefix skos: <http://www.w3.org/2004/02/skos/core#> .
@prefix xsd: <http://www.w3.org/2001/XMLSchema#> .

<http://codes.wmo.int/iwxxm/nil> rdfs:label "nil" ;
    dcterms:description "Nil" ;
    ns1:notation "nil" ;
    rdfs:member <http://codes.wmo.int/iwxxm/nil/AboveDetectionRange>,
        <http://codes.wmo.int/iwxxm/nil/BelowDetectionRange>,
        <http://codes.wmo.int/iwxxm/nil/inapplicable>,
        <http://codes.wmo.int/iwxxm/nil/missing>,
        <http://codes.wmo.int/iwxxm/nil/noSignificantChange>,
        <http://codes.wmo.int/iwxxm/nil/notDetectedByAutoSystem>,
        <http://codes.wmo.int/iwxxm/nil/notObservable>,
        <http://codes.wmo.int/iwxxm/nil/nothingOfOperationalSignificance>,
        <http://codes.wmo.int/iwxxm/nil/template>,
        <http://codes.wmo.int/iwxxm/nil/unknown>,
        <http://codes.wmo.int/iwxxm/nil/withheld> ;
    owl:versionInfo 12 .

<http://codes.wmo.int/iwxxm/nil/AboveDetectionRange> a skos:Concept ;
    rdfs:label "Above detection range"@en ;
    dcterms:description "The value was above the detection range of the instrument used to estimate it."@en ;
    owl:versionInfo <http://icao.int/iwxxm/2025-2> ;
    skos:notation "AboveDetectionRange" .

<http://codes.wmo.int/iwxxm/nil/BelowDetectionRange> a skos:Concept ;
    rdfs:label "Below detection range"@en ;
    dcterms:description "The value was below the detection range of the instrument used to estimate it."@en ;
    owl:versionInfo <http://icao.int/iwxxm/2025-2> ;
    skos:notation "BelowDetectionRange" .

<http://codes.wmo.int/iwxxm/nil/inapplicable> a skos:Concept ;
    rdfs:label "Inapplicable"@en ;
    dc:source "ISO 19136:2007 Geographic information - Geography Markup Language sub-clause 8.2.3.1"@en ;
    dcterms:description "There is no value."@en ;
    rdfs:seeAlso <http://www.opengis.net/doc/gml#clause-8.2.3.1> ;
    owl:versionInfo <http://icao.int/iwxxm/2025-2> ;
    skos:notation "inapplicable" .

<http://codes.wmo.int/iwxxm/nil/missing> a skos:Concept ;
    rdfs:label "Missing"@en ;
    dc:source "ISO 19136:2007 Geographic information - Geography Markup Language sub-clause 8.2.3.1"@en ;
    dcterms:description "No correct value is not readily available to the sender of this data. Furthermore, a correct value may not exist."@en ;
    rdfs:seeAlso <http://www.opengis.net/doc/gml#clause-8.2.3.1> ;
    owl:versionInfo <http://icao.int/iwxxm/2025-2> ;
    skos:notation "missing" .

<http://codes.wmo.int/iwxxm/nil/noSignificantChange> a skos:Concept ;
    rdfs:label "No significant change"@en ;
    dcterms:description "No significant change is expected to occur."@en ;
    owl:versionInfo <http://icao.int/iwxxm/2025-2> ;
    skos:altLabel "NOSIG"@en ;
    skos:notation "noSignificantChange" .

<http://codes.wmo.int/iwxxm/nil/notDetectedByAutoSystem> a skos:Concept ;
    rdfs:label "Nothing detected by automated system"@en ;
    dcterms:description "The automated observing system did not detect a value (e.g. no cloud detected NCD)."@en ;
    owl:versionInfo <http://icao.int/iwxxm/2025-2> ;
    skos:notation "notDetectedByAutoSystem" .

<http://codes.wmo.int/iwxxm/nil/notObservable> a skos:Concept ;
    rdfs:label "Not observable"@en ;
    dcterms:description "A system failure, sensor failure, or sensor obstruction, prevented the intended observation of the value."@en ;
    owl:versionInfo <http://icao.int/iwxxm/2025-2> ;
    skos:notation "notObservable" .

<http://codes.wmo.int/iwxxm/nil/nothingOfOperationalSignificance> a skos:Concept ;
    rdfs:label "Nothing of operational signficance"@en ;
    dcterms:description "Nothing was observed or forecast of operational significance (e.g. nil significant cloud NSC, nil significant weather NSW)."@en ;
    owl:versionInfo <http://icao.int/iwxxm/2025-2> ;
    skos:notation "nothingOfOperationalSignificance" .

<http://codes.wmo.int/iwxxm/nil/template> a skos:Concept ;
    rdfs:label "Template"@en ;
    dc:source "ISO 19136:2007 Geographic information - Geography Markup Language sub-clause 8.2.3.1"@en ;
    dcterms:description "The value will be available later."@en ;
    rdfs:seeAlso <http://www.opengis.net/doc/gml#clause-8.2.3.1> ;
    owl:versionInfo <http://icao.int/iwxxm/2025-2> ;
    skos:notation "template" .

<http://codes.wmo.int/iwxxm/nil/unknown> a skos:Concept ;
    rdfs:label "Unknown"@en ;
    dc:source "ISO 19136:2007 Geographic information - Geography Markup Language sub-clause 8.2.3.1"@en ;
    dcterms:description "The correct value is not known to, and not computable by, the sender of this data. However, a correct value probably exists."@en ;
    rdfs:seeAlso <http://www.opengis.net/doc/gml#clause-8.2.3.1> ;
    owl:versionInfo <http://icao.int/iwxxm/2025-2> ;
    skos:notation "unknown" .

<http://codes.wmo.int/iwxxm/nil/withheld> a skos:Concept ;
    rdfs:label "Withheld"@en ;
    dc:source "ISO 19136:2007 Geographic information - Geography Markup Language sub-clause 8.2.3.1"@en ;
    dcterms:description "The value is not divulged."@en ;
    rdfs:seeAlso <http://www.opengis.net/doc/gml#clause-8.2.3.1> ;
    owl:versionInfo <http://icao.int/iwxxm/2025-2> ;
    skos:notation "withheld" .
```

Needs to look deeper into check_url.py.

blchoy replied:
I have fixed makeIWXXMEntities.py and check_urls.py and the following is the error output for container nil.ttl:

```
======================================================================
FAIL: test_consistent_iwxxm_nil (__main__.TestContentsConsistency.test_consistent_iwxxm_nil)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/mnt/iwxxm-codelists/scripts/check_urls.py", line 166, in entity_consistent
    self.check_result(result_rdfgraph, expected_rdfgraph, uploads, identityURI, resourceURI)
  File "/mnt/iwxxm-codelists/scripts/check_urls.py", line 86, in check_result
    self.assertTrue(rdflib.compare.isomorphic(result, expected),
AssertionError: False is not true : https://ci.codes.wmo.int/iwxxm/nil

####### inBothinputs #######
@prefix rdfs: <http://www.w3.org/2000/01/rdf-schema#> .

<http://codes.wmo.int/iwxxm/nil> a <http://purl.org/linked-data/registry#Register>,
        <http://www.w3.org/ns/ldp#Container> ;
    rdfs:member <http://codes.wmo.int/iwxxm/nil/AboveDetectionRange>,
        <http://codes.wmo.int/iwxxm/nil/BelowDetectionRange>,
        <http://codes.wmo.int/iwxxm/nil/inapplicable>,
        <http://codes.wmo.int/iwxxm/nil/missing>,
        <http://codes.wmo.int/iwxxm/nil/noSignificantChange>,
        <http://codes.wmo.int/iwxxm/nil/notDetectedByAutoSystem>,
        <http://codes.wmo.int/iwxxm/nil/notObservable>,
        <http://codes.wmo.int/iwxxm/nil/nothingOfOperationalSignificance>,
        <http://codes.wmo.int/iwxxm/nil/template>,
        <http://codes.wmo.int/iwxxm/nil/unknown>,
        <http://codes.wmo.int/iwxxm/nil/withheld> .


####### inTestResultOnly #######
@prefix dc: <http://purl.org/dc/elements/1.1/> .
@prefix dcterms: <http://purl.org/dc/terms/> .
@prefix owl: <http://www.w3.org/2002/07/owl#> .
@prefix rdfs: <http://www.w3.org/2000/01/rdf-schema#> .
@prefix skos: <http://www.w3.org/2004/02/skos/core#> .

<http://codes.wmo.int/iwxxm/nil> a skos:Collection ;
    rdfs:label "Nil reasons"@en ;
    dc:source "Code Table D-1: Nil reasons"@en ;
    dcterms:description "Nil-reason terms are used to provide an explanation for recording a missing (or void) value within a data product. Terms are drawn from authorities in addition to WMO including ISO/TC 211 (from ISO 19136:2007 Geography markup language clause 8.2.3.1; published on their behalf by the Open Geospatial Consortium)."@en ;
    owl:versionInfo <http://icao.int/iwxxm/2025-2> .


####### inExpectedOnly #######
@prefix dcterms: <http://purl.org/dc/terms/> .
@prefix owl: <http://www.w3.org/2002/07/owl#> .
@prefix rdfs: <http://www.w3.org/2000/01/rdf-schema#> .
@prefix xsd: <http://www.w3.org/2001/XMLSchema#> .

<http://codes.wmo.int/iwxxm/nil> rdfs:label "nil" ;
    dcterms:description "Nil" ;
    owl:versionInfo 13 .
```

Looking good now.  However, I was not able to update the registry with the container TTL.  Something requiring @amilan17 's advise.

blchoy committed:
Revised check_urls.py and uploadChanges.py.

blchoy committed:
Committed further changes to check_urls.py.

blchoy committed:
Committed revised makeIWXXMEntities.py to improve the generation of container TTL.

blchoy committed:
Committed revised makeIWXXMEntities.py with updated TTL and RDF files, as well as revised check_urls.py, to fix issues on checking the difference between container TTL files and their online counterparts.
